File: labs/lab_12/ortiz_damien/p01/mymath.cc
// file: /data/courses/ece_1111/2025_01_fall/labs/lab_12/ortiz_damien/p01/p01.cc/
//
// Version History: 
// 20251110 create initial file

// include files 
//
#include "mymatrix.h"

// Program will read a matrix from a file, store this in a class called matrix,
// and then allow the user to call functions from the class to display and
// perform operations
//

// main program
//
int main(int argc, char** argv){

  // check that we have proper number of arguments
  //
  if(argc == 5){

    // create and read our matrices exiting
    // if they dont exist
    //
    MyMatrix matrix, matrix2;
    if(!matrix.read(argv[3])){
      return 1;
    }
    
    
    if(!matrix2.read(argv[4])){
      return 1;
    }

    // check that they tried to do a valid operation
    //
    if(OPERATOR.count(argv[1])){
      if (OPTIONS.count(argv[2])){

	// print results if operations are valid
	//
	fprintf(stdout, "First Matrix:\n");
	matrix.display(stdout);
	fprintf(stdout, "Second Matrix:\n");
	matrix2.display(stdout);
	fprintf(stdout, "Result:\n");
	performOp(argv[2], matrix, matrix2);
	
      }
      else{

	// tell user that there was an invalid operation
	//
	fprintf(stderr, "Undefined operation selected.\n");
	fprintf(stderr, "Usage: p01.exe -operation/-o <operation> <matrix_file> <matrix_file>\n");
	return 1;
	
      }
    } else{

      // tell user that there was an no valid option
      //
      fprintf(stderr, "Undefined option selected.\n");
      fprintf(stderr, "Usage: p01.exe -operation/-o <operation> <matrix_file> <matrix_file>\n");
      return 1;
      
    }

    // exit gracefully
    //
    return 0;
  }

  // tell user error when there are incorrect number of argument
  //
  else{
    
    fprintf(stderr, "Incorrect number of arguments.\n ");
    fprintf(stderr, "Usage: p01.exe -operation/-o <operation> <matrix_file> <matrix_file>\n");
    return 1;
    
  }
   
}

// end of file

File: labs/lab_12/ortiz_damien/p02/mymath.py
#!/usr/bin/env python

# file: /data/courses/ece_1111/2025_01_fall/labs/lab_12/ortiz_damien/p02/mymath/mymath.py
#

# Version History: 
# 20251111 create initial file

# Imports:
#
import mat_00
import sys
import os
import numpy as np

# main program
#
def main(argv):

        # create set of possible operations
        #
        s = set(["addition", "subtraction", "multiplication"])

        # check number of arguments is correct
        #
        if len(argv) == 5:

                # check that we called for an operation
                #
                if "-o" in argv[1] or "-operation" in argv[1]:

                        # check that operation is valid
                        #
                        if argv[2] in s:

                                # create our two matrices
                                #
                                mat1 = mat_00.MyMatrix()
                                mat2 = mat_00.MyMatrix()

                                # try to open files and perform operations
                                # 
                                if mat1.read(argv[3]):
                                        if mat2.read(argv[4]):

                                                # print matrices if all successful
                                                #
                                                mat3 = mat_00.performOp(argv[2], mat1, mat2)
                                                print("First matrix:")
                                                mat1.display()
                                                print("Second matrix:")
                                                mat2.display()
                                                print("Result matrix:")
                                                mat3.display()

                        else:
                                # tells user when there is a problem with operation
                                #
                                print("Invalid operation")
                                print("Usage: mymath.py -o <operation> <matrix_file> <matrix_file>")                                        
                else:
                        # tells user when there is a problem with option
                        #
                        print("Invalid options")
                        print("Usage: mymath.py -o <operation> <matrix_file> <matrix_file>")
                
                # exit gracefully
                #
                return True
        else:
                # tell user when wrong number of arguments
                #
                print("Incorrect number of arguments.")
                print("Usage: mymath.py -o <operation> <matrix_file> <matrix_file>")

                # exit with errors
                #
                return False
                      
                      
# begin gracefully

if __name__ == '__main__':
	main(sys.argv[0:])
        
# end of file
